core/server/webhooks-events.js

```diff
@@ -0,0 +1,79 @@
+var _ = require('lodash'),
+    errors = require('./errors'),
+    events = require('./events'),
+    logging = require('./logging'),
+    models = require('./models'),
+    https = require('https'),
+    url = require('url');
+
+function makeRequest(webhook, payload) {
+    var reqOptions, reqPayload, req;
+
+    reqOptions = url.parse(webhook.get('target_url'));
+    reqOptions.method = 'POST';
+    reqOptions.headers = {'Content-Type': 'application/json'};
+
+    reqPayload = JSON.stringify(payload);
+
+    logging.info('webhook.trigger', webhook.get('event'), webhook.get('target_url'));
+    req = https.request(reqOptions);
+
+    req.write(reqPayload);
+    req.on('error', function (err) {
+        // when a webhook responds with a 410 Gone response we should remove the hook
+        if (err.status === 410) {
+            logging.info('webhook.destroy (410 response)', webhook.get('event'), webhook.get('target_url'));
+            return models.Webhook.destroy({id: webhook.get('id')});
+        }
+
+        // TODO: use i18n?
+        logging.error(new errors.GhostError({
+            err: err,
+            context: {
+                id: webhook.get('id'),
+                event: webhook.get('event'),
+                target_url: webhook.get('target_url')
+            }
+        }));
+    });
+    req.end();
+}
+
+function triggerSubscriberAdded(subscriber) {
+    // webhook POST body should match GET request for same resource
+    // TODO: is `options` needed here?
+    var payload = {
+        subscribers: [subscriber.toJSON()]
+    };
+
+    // find relevant Webhooks
+    models.Webhook.findAllByEvent('subscriber.added').then(function (result) {
```

Comment:
This breaks a fundamental (and of course, undocumented) rule of the code base - we don't use the model layer outside of the API (with the exception of very specialist tooling like the importer). 

I get that this is an odd case, I think the thing that is most similar to this whole pattern is the mail api api/mail.js + the mail service that currently lives in server/mail  - even that's not the best example of a structure.